src/components/lib/cos_kernel_api.c

```diff
@@ -300,26 +303,78 @@ __page_bump_valloc(struct cos_compinfo *ci)
 		if (call_cap_op(meta->captbl_cap, CAPTBL_OP_PGTBLACTIVATE,
 				pte_cap, meta->mi.pgtbl_cap, ptemem_cap, 1)) {
 			assert(0); /* race? */
```

Comment:
I was thinking that returns after assert(0) aren't useful but after I saw your other comment below, I realized my mistake. I'll put them back. Sorry.